crates/viewer/re_component_ui/src/datatype_uis/vec.rs

```diff
@@ -12,50 +25,80 @@ pub fn edit_or_view_vec3d(
+    let speed = (value.abs() * 0.01).at_least(0.001);
+    egui::DragValue::new(value)
+        .clamp_existing_to_range(false)
+        .range(range)
+        .speed(speed)
+        .suffix(suffix)
 }
 
-fn edit_or_view_vector_component_immutable(
+// TODO(#6743): Since overrides are not yet taken into account, editing this value has no effect.
+pub fn edit_or_view_vec2d_raw(
     ui: &mut egui::Ui,
-    value: &mut MaybeMutRef<'_, datatypes::Vec3D>,
-    i: usize,
+    value: &mut MaybeMutRef<'_, datatypes::Vec2D>,
 ) -> egui::Response {
-    let mut value: MaybeMutRef<'_, f32> = match value {
-        MaybeMutRef::Ref(value) => MaybeMutRef::Ref(&value[i]),
+    let x = value.0[0];
+    let y = value.0[1];
 
-        MaybeMutRef::MutRef(value) => MaybeMutRef::Ref(&value[i]),
-    };
-    edit_f32_float_raw(ui, &mut value, f32::MIN..=f32::MAX, "")
+    if let Some(value) = value.as_mut() {
+        let mut x_edit = x;
+        let mut y_edit = y;
+
+        let response_x = ui.add(drag(&mut x_edit, f32::MIN..=f32::MAX, ""));
+        let response_y = ui.add(drag(&mut y_edit, f32::MIN..=f32::MAX, ""));
+
+        let response = response_y | response_x;
+
+        if response.changed() {
+            *value = datatypes::Vec2D([x_edit, y_edit]);
+        }
+
+        response
+    } else {
+        ui.label(format!(
+            "[ {} , {} ]",
+            re_format::format_f32(x),
+            re_format::format_f32(y),
+        ))
+    }
 }
 
+// TODO(#6743): Since overrides are not yet taken into account, editing this value has no effect.
 pub fn edit_or_view_vec3d_raw(
     ui: &mut egui::Ui,
     value: &mut MaybeMutRef<'_, datatypes::Vec3D>,
 ) -> egui::Response {
-    edit_or_view_vector_component(ui, value, 0)
-        | edit_or_view_vector_component(ui, value, 1)
-        | edit_or_view_vector_component(ui, value, 2)
-}
+    let x = value.0[0];
+    let y = value.0[1];
+    let z = value.0[2];
 
-fn edit_or_view_vector_component(
-    ui: &mut egui::Ui,
-    value: &mut MaybeMutRef<'_, datatypes::Vec3D>,
-    i: usize,
-) -> egui::Response {
-    let mut value: MaybeMutRef<'_, f32> = match value {
-        MaybeMutRef::Ref(value) => MaybeMutRef::Ref(&value[i]),
-        MaybeMutRef::MutRef(value) => MaybeMutRef::MutRef(&mut value[i]),
-    };
-    edit_f32_float_raw(ui, &mut value, f32::MIN..=f32::MAX, "")
+    if let Some(value) = value.as_mut() {
+        let mut x_edit = x;
+        let mut y_edit = y;
+        let mut z_edit = z;
+
+        let response_x = ui.add(drag(&mut x_edit, f32::MIN..=f32::MAX, ""));
+        let response_y = ui.add(drag(&mut y_edit, f32::MIN..=f32::MAX, ""));
+        let response_z = ui.add(drag(&mut z_edit, f32::MIN..=f32::MAX, ""));
+
+        let response = response_y | response_x | response_z;
+
+        if response.changed() {
+            *value = datatypes::Vec3D([x_edit, y_edit, z_edit]);
+        }
+
+        response
+    } else {
+        ui.label(format!(
+            "[ {} , {}, {} ]",
```

Comment:
```suggestion
            "[ {}, {}, {} ]",
```

... or is it?

```suggestion
            "[ {} , {} , {} ]",
```
that's what the 2d version uses :)